modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -231,6 +231,20 @@ public ReturnType call(final String[] args, Cfg cfg) {
 
             // 5. options that can be influenced by the -d and -n arguments
 
+            if (options.getPort() != null) {
+                int port_number = Integer.parseInt(options.getPort());
+                if (port_number < 0 || port_number > 0xFFFF) {
+                    System.out.println(
+                            "\nport out of range: "
+                                    + port_number
+                                    + ", switching to the current port configuration: \n");
+                } else {
+                    // update port in the config file
+                    cfg.toXML(new String[] {"--p2p=" + "," + options.getPort()}, configFile);
```

Comment:
This line will modify the initial config file. You can verify that `config/mainnet/config.xml` will have the value of the port. It also ends up writing the id in the initial file, which is not the desired outcome. To avoid these side-effects you can set the port value inside the `cfg` object and set the `overwrite` flag to `true`. Writing to file will be triggered correctly below due to the `overwrite` flag.

Additionally, you can see in the tests (e.g. options `-p 12345`) that the config file is overwritten twice (not necessary).

Reply:
:+1: